content/articles/How-to-start-up-with-Socket-Programming-with-Java/index.md

```diff
@@ -0,0 +1,331 @@
+
+### What is a Socket?
+A socket is an **endpoint** of communication, acting as a connection between two computers on a network. It contains a socket number that the TCP layer can use to identify the application to which data will be transmitted. An endpoint is made up of a port number and an IP address.
+
+There are two types of sockets
+
+1. Server socket - which waits for a request from clients
+2. Client socket - can be used to send and receive a request from clients.
+
+The client must know two information about the server:
+1.IP address of Server and 
+2.Port number
+
+>Port number 0-1024 are reserved for administrative purposes(e,g 21 for FTP,23 for Telnet,25 for email and 80 for HTTP).
+
+This happens when a TCP connection is being used between two machines:-
+
+1. server creates a ServerSocket object, to indicate which port number our communication is to occur. If an I/O  error occurs, exception handling mechanisms are implemented; **Throws** an **I/OException** or a **try** **catch** block is used.
+2. The accept() function of the ServerSocket class is called by the server. This method waits for a client to connect to the server on a specific port before returning.
+3. A client builds an object in the socket, giving the name of the server and the connecting port number as the server awaits.
+4. The Socket Class builder will try to connect the client to the server and port number provided. The client now has a socket object that, if communication is established, may connect with the server.
+5. The accept() method on the server produces a reference to a new server socket that is connected to the client's socket.
+ 
+If the connection becomes a success, communication can therefore be established using **I/O streams**. Each socket has an input and output stream. The client's OutputStream connects to an InputStream of the server while the server's OutputStream connects to the client's InputStream.
+
+A stream is a collection of data in a particular order. Character streams can be divided into two categories:
+1. Character stream (usually used with text files)
+2. Byte stream(usually used with images)
+Character stream is in human-readable language while byte stream is in machine language, a language understandable by the CPU.
+
+Our data/messages will be sent across both streams at the same time because TCP is a two-way communication protocol.
+   
+### Programming on Server-side 
+A server socket is an instance of ServerSocket class and can be created by one of the following constructors: 
+
+```java
+serversocket= new Serversocket ( int port )
+```
+
+port: port number at which the server will be listening for requests from clients.
+
+The ServerSocket object sits and waits for network requests to arrive. Our client will try to connect to port 5000 that we created. 
```

Comment:
>Our client will try to connect to port 5000 that we created

This sentence is confusing to me because you never mentioned that we are using port 5000 and I'm not sure "created" the right word choice there, it would probably be something closer to "selected".